open-sphere-base/control-panels/src/main/java/io/opensphere/controlpanels/animation/view/AnimationOptionsProvider.java

```diff
@@ -59,19 +65,82 @@ public JPanel getOptionsPanel()
                     "Do not allow the loop span to be changed by the active span");
             ToolTipManager.sharedInstance().setDismissDelay(10000);
 
-            JButton resetFrameButton = new JButton("Reset timeline size & location");
-            resetFrameButton.addActionListener(e -> myTimelineFrame.resizeAndPositionToDefault());
-
             ViewPanel panel = new ViewPanel();
-            panel.addComponent(ControllerFactory.createComponent(myAnimationModel.getRememberTimes()));
-            panel.addLabelComponent(ControllerFactory.createLabelAndComponent(myAnimationModel.getViewPreference()));
-            panel.addComponent(ControllerFactory.createComponent(myAnimationModel.getLoopSpanLocked()));
-            panel.addComponent(resetFrameButton);
+            panel.addComponent(getRememberLoopSpanCheckBox());
+            panel.addLabelComponent(getTimelineViewPreferenceComponents());
+            panel.addComponent(getLockLoopSpanCheckBox());
+            panel.addComponent(getResetFrameButton());
             myPanel = new OptionsPanel(panel);
         }
         return myPanel;
     }
 
+    /**
+     * Gets the remember loop span checkbox.
+     *
+     * @return the remember loop span checkbox
+     */
+    private Component getRememberLoopSpanCheckBox()
+    {
+        JComponent rememberLoopSpan = ControllerFactory.createComponent(myAnimationModel.getRememberTimes());
+        if (rememberLoopSpan instanceof JCheckBox)
+        {
+            JCheckBox rememberLoopSpanCheckBox = (JCheckBox)rememberLoopSpan;
+            rememberLoopSpanCheckBox.addActionListener(e -> Quantify.collectEnableDisableMetric(
+                    "mist3d.settings.timeline-and-animation.remember-loop-span", rememberLoopSpanCheckBox.isSelected()));
+        }
+        return rememberLoopSpan;
+    }
+
+    /**
+     * Gets the timeline view preference components - label and combobox.
+     *
+     * @return the timeline view preference components
+     */
+    private JComponent[] getTimelineViewPreferenceComponents()
+    {
+        JComponent[] timelineViewPreference = ControllerFactory.createLabelAndComponent(myAnimationModel.getViewPreference());
+        if (timelineViewPreference[1] instanceof JComboBox<?>)
+        {
+            ((JComboBox<?>)timelineViewPreference[1]).addActionListener(
+                e -> Quantify.collectMetric("mist3d.settings.timeline-and-animation.timeline-to-show-on-startup-selection"));
+        }
+        return timelineViewPreference;
+    }
+
+    /**
+     * Gets the lock loop span checkbox.
+     *
+     * @return the lock loop span checkbox
+     */
+    private Component getLockLoopSpanCheckBox()
+    {
+        JComponent lockLoopSpan = ControllerFactory.createComponent(myAnimationModel.getLoopSpanLocked());
+        if (lockLoopSpan instanceof JCheckBox)
+        {
```

Comment:
As this is a boolean model, it will always be a JCheckBox. We can ditch the wrapping if statement.

Reply:
Done